core/src/main/java/hudson/Util.java

```diff
@@ -1186,6 +1179,22 @@ public static void createSymlink(@Nonnull File baseDir, @Nonnull String targetPa
             Path path = fileToPath(new File(baseDir, symlinkPath));
             Path target = Paths.get(targetPath, MemoryReductionUtil.EMPTY_STRING_ARRAY);
 
+            try {
+                File symlink = File.createTempFile("tmp-"+symlinkPath, null, baseDir);
+                symlink.delete();
+                Path tempSymlinkPath = symlink.toPath();
+                Files.createSymbolicLink(tempSymlinkPath, target);
+                Files.move(tempSymlinkPath, path, StandardCopyOption.ATOMIC_MOVE);
+                return;
+            } catch (AtomicMoveNotSupportedException e) {
```

Comment:
Catching everything without even no log does not seem fine.

Reply:
If it fails, it just takes the older path. There's no reason to log.

Reply:
Even like that, I think logging something about the error does not hurt.

Reply:
You're wrong. Jenkins defaults to complaining about a missing ALPN module. It hints that if you feed it one, it will shut up. So we fed it one, and it was blissfully quiet for months. And then eventually a java upgrade resulted in the ALPN module breaking SSL/SSH entirely, making Jenkins absolutely non-functional.

But before you respond, please talk to @daniel-beck because he helped me debug that.

Logging should not be used if the code will work perfectly fine and the condition is expected. Especially if there isn't anything useful that one can do about it. There could for instance be a file system that doesn't support MOVE_ATOMIC or a JRE that doesn't support it. Just because those edges may exist doesn't mean we shouldn't try to use the feature.